api/src/main/java/com/flipkart/poseidon/validator/TaskValidator.java

```diff
@@ -34,14 +40,35 @@
         for (Map.Entry<String, TaskPOJO> entry : tasks.entrySet()) {
             String taskName = entry.getKey();
             TaskPOJO task = entry.getValue();
+            final Class<? extends DataSource<?>> dsClass = datasources.get(task.getName());
```

Comment:
There is a null or empty check after this line. What happens when null is passed to map?

Reply:
it'll be null